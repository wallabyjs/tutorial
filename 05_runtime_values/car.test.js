import Car from './car';

let car;

describe('car', () => {

  beforeEach(() => {
    car = new Car(88);
  });

  it('has correct engine', () => {
    const engine = car.engine;
    expect(engine).toMatchSnapshot();
  });

  it('brakes', () => {
    car.brake();
    expect(car.speed).toBe(87);
  });
});