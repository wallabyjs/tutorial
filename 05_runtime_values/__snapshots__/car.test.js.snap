// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`car has correct engine 1`] = `
{
  "spec": {
    "aspiration": "Twin Turbocharged",
    "bore": "93mm",
    "compression_ratio": "7.5:1",
    "cooling_system": "Water-cooled",
    "displacement": "2.85L",
    "engine_code": "ZMJ-159-TT",
    "exhaust_system": "Stainless Steel Exhaust",
    "firing_order": "1-6-3-5-2-4",
    "fuel_system": "Bosch K-Jetronic Fuel Injection",
    "fuel_type": "Gasoline",
    "horsepower": 250,
    "ignition_system": "Electronic Ignition",
    "make": "DeLorean",
    "model": "PRV ZMJ-158 Twin Turbo (Aftermarket)",
    "modifications": {
      "fuel_injectors": "High-flow Fuel Injectors",
      "intercooler": "Air-to-Air Intercooler",
      "tuning": "Custom ECU Tune",
    },
    "redline": 7500,
    "stroke": "73mm",
    "torque": {
      "unit": "lb-ft",
      "value": 285,
    },
    "turbocharger": {
      "boost_pressure": {
        "unit": "psi",
        "value": 7,
      },
      "brand": "Garrett",
    },
    "type": "V6",
    "valvetrain": "SOHC",
    "weight": {
      "unit": "lbs",
      "value": 375,
    },
  },
  "volume": "2.85 L",
}
`;
